### PR TITLE
Tokenize

### PR DESCRIPTION
**Pull request by aishwmano**
Added working tokenize functionality

### CODE REVIEW

File: src/codeu/chat/util/Tokenizer.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package codeu.chat.util;
+
+import java.io.*;
+import java.io.IOException;
+import java.lang.*;
+
+public final class Tokenizer {
```
Comment (cbick) [inline src/codeu/chat/util/Tokenizer.java]: the class and its public methods could use some documentation

File: src/codeu/chat/util/Tokenizer.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  private StringBuilder token;
```
Comment (cbick) [inline src/codeu/chat/util/Tokenizer.java]: token and source should be final, i think

File: test/codeu/chat/util/TokenizerTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    assertEquals(tokenizer.next(), null);
+  }
+  @Test
+  public void testWithNoQuotes() throws IOException {
+    final Tokenizer tokenizer = new Tokenizer("\"hello world\" \"how are you\"");
+    assertEquals(tokenizer.next(), "hello world");
+    assertEquals(tokenizer.next(), "how are you");
+    assertEquals(tokenizer.next(), null);
+   }
+}
```
Comment (cbick) [inline test/codeu/chat/util/TokenizerTest.java]: test imbalanced quotes, mixture of quoted/nonquoted?

Comment (oldmud0) [reply test/codeu/chat/util/TokenizerTest.java]: Don't forget Unicode.

File: src/codeu/chat/client/commandline/Chat.java [hunk 4]
```diff
@@ -18,11 +18,14 @@
 import java.util.Map;
 import java.util.Scanner;
 import java.util.Stack;
+import java.io.*;
+import java.util.*;
```
Comment (oldmud0) [inline src/codeu/chat/client/commandline/Chat.java]: Don't do wildcard imports; they cause confusion in many cases. Let Eclipse take care of imports for you.

File: src/codeu/chat/client/commandline/Chat.java [hunk 5]
```diff
@@ -45,11 +48,15 @@ public Chat(Context context) {
...
-    final Scanner tokens = new Scanner(line.trim());
-
-    final String command = tokens.hasNext() ? tokens.next() : "";
+    final List<String> args = new ArrayList<String>();
+    Tokenizer tokenizer = new Tokenizer(line);
+    for (String token = tokenizer.next(); token != null; token = tokenizer.next()) {
+      args.add(token);
+    }
+    final String command = args.get(0);
+    args.remove(0);
```
Comment (oldmud0) [inline src/codeu/chat/client/commandline/Chat.java]: [`List.remove`](http://docs.oracle.com/javase/8/docs/api/java/util/List.html#remove-int-) also returns the element at that position before it was removed, so you can condense these two lines into one:
```java
final String command = args.remove(0);
```

(edit: looks like you already know this, as shown by later code, so I'm left scratching my head as to why this is here.)

File: src/codeu/chat/util/Tokenizer.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
+package codeu.chat.util;
+
+import java.io.*;
+import java.io.IOException;
+import java.lang.*;
```
Comment (oldmud0) [inline src/codeu/chat/util/Tokenizer.java]: Remove wildcard imports.

File: src/codeu/chat/util/Tokenizer.java (hunk 1 above)
Comment (oldmud0) [reply src/codeu/chat/util/Tokenizer.java]: I second this.

File: src/codeu/chat/util/Tokenizer.java (hunk 2 above)
Comment (oldmud0) [reply src/codeu/chat/util/Tokenizer.java]: Indeed.

**Review by oldmud0 — CHANGES_REQUESTED** [review]
This big red 'X' looks very imposing, but I just need you to address those changes before I merge everything, as the deadline is approaching rather quickly.

**Comment by aishwmano** [comment]
Made the suggested changes.

**Pull request merged**